modules/cpmstarBidAdapter.js

```diff
@@ -0,0 +1,121 @@
+
+  getMediaType: function(bidRequest) {
+    if (bidRequest == null) return BANNER;
+    return !utils.deepAccess(bidRequest, 'mediaTypes.video') ? BANNER : VIDEO;
+  },
+
+  getPlayerSize: function(bidRequest) {
+    var playerSize = utils.deepAccess(bidRequest, 'mediaTypes.video.playerSize');
+    if (playerSize == null) return [640, 440];
+    if (playerSize[0] != null) playerSize = playerSize[0];
+    if (playerSize == null || playerSize[0] == null || playerSize[1] == null) return [640, 440];
+    return playerSize;
+  },
+
+  isBidRequestValid: function (bid) {
+    return ((typeof bid.params.placementId === 'string') && !!bid.params.placementId.length) || (typeof bid.params.placementId === 'number');
+  },
+
+  buildRequests: function (validBidRequests, bidderRequest) {
+    var requests = [];
+    // This reference to window.top can cause issues when loaded in an iframe if not protected with a try/catch.
+
+    for (var i = 0; i < validBidRequests.length; i++) {
+      var bidRequest = validBidRequests[i];
+      var referer = encodeURIComponent(bidderRequest.refererInfo.referer);
+      var e = utils.getBidIdParameter('endpoint', bidRequest.params);
+      var ENDPOINT = e == 'dev' ? ENDPOINT_DEV : e == 'staging' ? ENDPOINT_STAGING : ENDPOINT_PRODUCTION;
+      var mediaType = spec.getMediaType(bidRequest);
+      var playerSize = spec.getPlayerSize(bidRequest);
+      var videoArgs = '&fv=0' + (playerSize ? ('&w=' + playerSize[0] + '&h=' + playerSize[1]) : '');
+      requests.push({
+        method: 'GET',
+        url: ENDPOINT + '?media=' + mediaType + (mediaType == VIDEO ? videoArgs : '') + '&json=c_b&mv=1&poolid=' + utils.getBidIdParameter('placementId', bidRequest.params) + '&reachedTop=' + encodeURIComponent(bidderRequest.refererInfo.reachedTop) + '&requestid=' + bidRequest.bidId + '&referer=' + referer,
+        bidRequest: bidRequest,
+      });
+    }
+
+    return requests;
+  },
+
+  onTimeout(timeout) {
+  },
+  onBidWon(winningBid) {
+  },
+
+  interpretResponse: function (serverResponse, request) {
+    var bidRequest = request.bidRequest;
+    var mediaType = spec.getMediaType(bidRequest);
+
+    var bidResponses = [];
+
+    if (!Array.isArray(serverResponse.body)) {
+      serverResponse.body = [serverResponse.body];
+    }
+
+    for (var i = 0; i < serverResponse.body.length; i++) {
+      var raw = serverResponse.body[i];
+      var rawBid = raw.creatives[0];
+      if (!rawBid) {
+        return;
+      }
+      var cpm = (parseFloat(rawBid.cpm) || 0);
+
+      if (!cpm) {
+        return;
+      }
+
+      var bidResponse = {
+        requestId: rawBid.requestid,
+        cpm: cpm,
+        width: rawBid.width || 0,
+        height: rawBid.height || 0,
+        currency: rawBid.currency ? rawBid.currency : DEFAULT_CURRENCY,
+        netRevenue: rawBid.netRevenue ? rawBid.netRevenue : true,
+        ttl: rawBid.ttl ? rawBid.ttl : DEFAULT_TTL,
+        creativeId: rawBid.creativeid || 0,
+        ad: ''
```

Comment:
Given you're supporting both video and banner bids, I'd recommend to remove this default value for the `ad` property.  The `ad` is only used in the `banner` type, so it would be left equal to an empty string when you just serve back a video bid.  

Having both fields present in the bid may cause some confusion, so it'd be better to only assign the property when its needed.

Reply:
This has been removed -- thanks for the tip.